### ISSUE TITLE
Cannot Download osx-razer-led

### ISSUE THREAD

**Issue by wu-hongjun** [BlvckBytes/RazerControl#2]
Hi,

I tried to run the program on my Razer Blade 15 Adv mid2019 and it pops this error:
![Screen Shot 2020-06-28 at 10 15 05 PM](https://user-images.githubusercontent.com/27667864/85975465-f9755e00-b98c-11ea-9a21-504711bf82cc.png)
![Screen Shot 2020-06-28 at 10 15 22 PM](https://user-images.githubusercontent.com/27667864/85975471-fc704e80-b98c-11ea-9f1a-0504d3060807.png)

I am not sure what to do now...

**Comment by BlvckBytes** [comment]
Hey there!

Could you please provide me with the whole error message from the menubar
item? Where the scrollbar appeared, just copy and paste it in here. I don't
know why the curl failed for you there otherwise.

Best Regards,
BlvckBytes

Hongjun Jack Wu <notifications@github.com> schrieb am Mo., 29. Juni 2020,
07:16:

**Comment by wu-hongjun** [comment]
> Hey there! Could you please provide me with the whole error message from the menubar item? Where the scrollbar appeared, just copy and paste it in here. I don't know why the curl failed for you there otherwise. Best Regards, BlvckBytes Hongjun Jack Wu <notifications@github.com> schrieb am Mo., 29. Juni 2020, 07:16:
> […](#)

I was unable to select text in that window so I just snipped many pictures.
![Screen Shot 2020-06-28 at 10 32 38 PM](https://user-images.githubusercontent.com/27667864/85976757-0e072580-b990-11ea-80bf-d600c2cb9cfe.png)
![Screen Shot 2020-06-28 at 10 32 50 PM](https://user-images.githubusercontent.com/27667864/85976771-152e3380-b990-11ea-84fd-2b3ebe951c3a.png)
![Screen Shot 2020-06-28 at 10 32 58 PM](https://user-images.githubusercontent.com/27667864/85976782-1b241480-b990-11ea-981e-64425057faef.png)
![Screen Shot 2020-06-28 at 10 33 06 PM](https://user-images.githubusercontent.com/27667864/85976792-1fe8c880-b990-11ea-8035-3f6599adf859.png)

Hope this helps.

**Comment by BlvckBytes** [comment]
Oh, whoops, seems like I have disabled user-select in CSS there, sorry :,D.
To be honest, I don't quite get what issue curl is having there... Maybe it has no permissions to write at the destination location?

Please try the following:
```mkdir /usr/local/bin/test```

Does that give you any errors? If not, then you can remove it again.
```rmdir /usr/local/bin/test```

This could be a SIP-thing, which I don't yet believe... I have disabled SIP which could be why this always worked for me.

**Comment by vtashev** [comment]
Hi @BlvckBytes BlvckBytes 

So happy to found you outside tonymacosx86 :)

I'm in the same boat as @Errrneist 

I did disable SIP and mount / as rw before starting your app.
Do you have inside the osx-razer-led or shall I get the release from https://github.com/DocSystem/osx-razer-led ?

Cheers

**Comment by BlvckBytes** [comment]
@vtashev 

Hey there! :)
As you can see inside this sourcefile: https://github.com/BlvckBytes/RazerControl/blob/master/electron_modules/binaryDownloader.js

I just download the binary and move it to /usr/local/bin. You could try doing that by hand, and afterwards chmod +x the file. I really don't know why the auto-download doesn't work for some people, but it does for others. I'm quite short on time rn, so please try the manual method, since it's a one time thing.

Get the binary here: https://github.com/dylanparker/osx-razer-led/releases/download/v0.1/osx-razer-led

**Comment by vtashev** [comment]
Odly on my Catalina when I do
` mv osx-razer-led /usr/local/bin/
mv: rename osx-razer-led to /usr/local/bin/: Not a directory`

Am I missing something here?

**Comment by BlvckBytes** [comment]
@vtashev 

Could be that your bin in /usr/local does not yet exist. Try creating it first, then moving into this directory. Should be something like this:

```
mkdir /usr/local/bin
mv osx-razer-led /usr/local/bin
chmod +x /usr/local/bin/osx-razer-led
```

And you *may* need sudo to do so.

**Comment by vtashev** [comment]
Silly me!
I had a file for /usr/local/bin
Now it is working fine = thanks!

I have something else in mind: have you found how to change the default f1-f12 function from f's to volume/next/brig controls?
I did a profile under windows which light just the volume keys/kb brightness/lcd brightness and I trigger the profile with fn+4(just my case)

**Comment by BlvckBytes** [comment]
@vtashev 

I just use karabiner elements for that. You can search for the ZBOOK_HACK repo on my profile, there should be an image in README of how it looks like.

**Comment by BlvckBytes** [comment]
Thinking of it... It may be smarter to keep the files related to this app in ~/.rzrCtl

Well, maybe going to change that soon.

**Comment by Newfiejudd** [comment]
I have done the above mentioned update and the error as originally posted is no longer there. Although now it seems the app continually stays in the writing mode without actually making the system changes. 
- Sip Disabled in Config.plist
- NVRAM reset
- SIP also disabled via recovery

Just can't get the app to work correctly in the latest version Big Sur 11.2.3. Any help would be greatly appreciated. These RBA make excellent hacks and this is the only item I have to correct.

**Comment by BlvckBytes** [comment]
@Newfiejudd

Does the command line binary work for you? Like, try just setting a static color for example.

**Comment by Newfiejudd** [comment]
No I have no control over the app. 
![Screen Shot 2021-03-22 at 11 22 38 AM](https://user-images.githubusercontent.com/69902460/112000599-16e0ee00-8b01-11eb-844d-4c05bd095f58.png)

**Comment by BlvckBytes** [comment]
@Newfiejudd

No, I mean the command line tool, which this GUI only instructs. Type osx-razer-led into the terminal, then make use of it's output to figure out how to set a static color, and see if that works.

**Comment by Newfiejudd** [comment]
Sorry I had no luck with the command line also. 

 ~ % osx-razer-led static 255 255 255

**Comment by BlvckBytes** [comment]
@Newfiejudd

No output from executing this command?

**Comment by Newfiejudd** [comment]
No sir. 
retrying now. Just verified that SIP is fully disabled, along with the LED file in the correct location. 
![Screen Shot 2021-03-22 at 11 40 28 AM](https://user-images.githubusercontent.com/69902460/112003060-7e983880-8b03-11eb-8b60-9c4cb1802e3b.png)

**Comment by BlvckBytes** [comment]
@Newfiejudd

Can you send me a screenshot of your hackintool's USB map tab?

**Comment by Newfiejudd** [comment]
![Screen Shot 2021-03-22 at 11 43 40 AM](https://user-images.githubusercontent.com/69902460/112003473-e3539300-8b03-11eb-94e7-9c79a4db9b6a.png)

**Comment by Newfiejudd** [comment]
![Uploading Screen Shot 2021-03-22 at 11.44.35 AM.png…]()

**Comment by Newfiejudd** [comment]
![Screen Shot 2021-03-22 at 11 44 35 AM](https://user-images.githubusercontent.com/69902460/112003695-1433c800-8b04-11eb-8a98-c73a2569e788.png)

**Comment by Newfiejudd** [comment]
I think I know exactly where my issue is now. I will rerun the mapping and ensure I have the correct USB kext installed.

**Comment by BlvckBytes** [comment]
@Newfiejudd

Oh, you're on 11.2.3... Hmm, I only tested it on 11.1, and I don't own the razer blade anymore :/. Maybe macOS changed the USB permissions. You could try running that command as sudo, probably won't change much tho.

Yeah, try to make a proper USB map. But since "Razer Blade" is marked as internal already, I don't think that'll change much either.

**Comment by Newfiejudd** [comment]
Thanks again for the help. Much appreciated.
